testing/src/main/java/com/linecorp/decaton/testing/KafkaAdmin.java

```diff
@@ -63,10 +63,12 @@ public void createTopic(String topicName, int numPartitions, int replicationFact
     }
 
     public void deleteTopics(String... topicNames) {
-        try {
-            adminClient.deleteTopics(Arrays.asList(topicNames)).all().get();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+        if (!EmbeddedKafkaCluster.IS_WINDOWS) {
```

Comment:
> https://github.com/line/decaton/pull/69/files#diff-2d8a3a3355ba2b4c1d8f5e4395d91804R71 isn't this change enough to prevent kafka to perform physical deletion, so we can bypass the problem described in jira?

That change means the embedded kafka cluster throws an exception instead of performing deletion. So we still need to avoid calling the delete method somehow.

> Even if the target cluster is a locally running embedded cluster, I think to avoid failing build due to related tests on windows its better to skip the related test itself (conditionally ignore) on windows rather than disabling only the topic delete because the testing logic might depends on topic deletion. It's not desirable but at least we can still catch an issue at CI if the change introduced any undesired side effect.

In principle I agree with you, but the only place this method is actually called is as a "cleanup"-style deletion. So it seems a shame to disable all our tests on windows when actually they could run fine. How about if I add a flag e.g. `failSilently`, that can be `false` for cleanup and `true` if the test needs the deletion to happen as part of its semantics (in which case the test would have to be skipped on Windows).

Reply:
Okay that sounds like an option 👍